packages/firestore/test/integration/api/validation.test.ts

```diff
@@ -271,30 +301,41 @@ apiDescribe('Validation:', (persistence: boolean) => {
   describe('Document paths', () => {
     validationIt(persistence, 'must be strings', db => {
       const baseCollectionRef = db.collection('foo');
-      expect(() => db.doc(null as any)).to.throw(
-        'Function Firestore.doc() requires its first argument to be ' +
-          'of type non-empty string, but it was: null'
-      );
-      expect(() => db.doc('')).to.throw(
-        'Function Firestore.doc() requires its first argument to be ' +
-          'of type non-empty string, but it was: ""'
-      );
-      expect(() => baseCollectionRef.doc(null as any)).to.throw(
-        'Function CollectionReference.doc() requires its first ' +
-          'argument to be of type non-empty string, but it was: null'
-      );
-      expect(() => baseCollectionRef.doc('')).to.throw(
-        'Function CollectionReference.doc() requires its first ' +
-          'argument to be of type non-empty string, but it was: ""'
-      );
-      expect(() => baseCollectionRef.doc(undefined as any)).to.throw(
-        'Function CollectionReference.doc() requires its first ' +
-          'argument to be of type non-empty string, but it was: undefined'
-      );
-      expect(() => (baseCollectionRef.doc as any)('foo', 'bar')).to.throw(
-        'Function CollectionReference.doc() requires between 0 and ' +
-          '1 arguments, but was called with 2 arguments.'
-      );
+      if (usesModularApi()) {
```

Comment:
Since the only differences in the error messages are the function name and argument position, it seems like we could redefine these tests as the common set and those that are additional for the classic API. As it stands there's nothing requiring any of the equivalent validation checks to be identical. Given their superficial dissimilarity it seems possible for someone to overlook that a message needs to be updated in both places.

I'm not actually sure this is a good idea, but I thought I'd mention the idea to see if it resonates with you at all.

Reply:
Done. This should also go away as we productionize the shim.